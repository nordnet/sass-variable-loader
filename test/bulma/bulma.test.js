const { readFileSync } = require('fs');
const pathBuilder = require('path');
const rootDir = require('../../root-dir');
const parseVariables = require('../../src/parse-variables');

const bulmaRoot = pathBuilder.join(rootDir, 'node_modules', 'bulma', 'sass');

function getDerivedVariablesSass(file) {
  const content = readFileSync(file, 'utf8');
  return [
    '@import "initial-variables.sass"',
    '@import "functions.sass"',
    '',
    content,
  ].join('\n');
}

describe('bulma', () => {
  test('derived-variables.sass should match a snapshot', () => {
    const cwd = pathBuilder.join(bulmaRoot, 'utilities');
    const sass = getDerivedVariablesSass(
      pathBuilder.join(cwd, 'derived-variables.sass'),
    );
    const variables = parseVariables(sass, { indented: true, cwd });

    expect(typeof variables).toBe('object');
    expect(Object.keys(variables).length).toBeGreaterThan(20);
    expect(variables).toMatchSnapshot();
  });
});
