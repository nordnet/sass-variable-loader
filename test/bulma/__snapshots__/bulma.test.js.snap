// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bulma derived-variables.sass should yield a large object 1`] = `
Object {
  "background": "whitesmoke",
  "blueInvert": "#fff",
  "border": "#dbdbdb",
  "borderHover": "#b5b5b5",
  "code": "#ff3860",
  "codeBackground": "whitesmoke",
  "cyanInvert": "#fff",
  "danger": "#ff3860",
  "dangerInvert": "#fff",
  "dark": "#363636",
  "darkInvert": "whitesmoke",
  "familyCode": "monospace",
  "greenInvert": "#fff",
  "info": "#209cee",
  "infoInvert": "#fff",
  "light": "whitesmoke",
  "lightInvert": "#363636",
  "link": "#3273dc",
  "linkActive": "#363636",
  "linkActiveBorder": "#4a4a4a",
  "linkFocus": "#363636",
  "linkFocusBorder": "#3273dc",
  "linkHover": "#363636",
  "linkHoverBorder": "#b5b5b5",
  "linkInvert": "#fff",
  "linkVisited": "#b86bff",
  "orangeInvert": "#fff",
  "pre": "#4a4a4a",
  "preBackground": "whitesmoke",
  "primary": "#00d1b2",
  "primaryInvert": "#fff",
  "purpleInvert": "#fff",
  "redInvert": "#fff",
  "sizeLarge": "1.5rem",
  "sizeMedium": "1.25rem",
  "sizeNormal": "1rem",
  "sizeSmall": "0.75rem",
  "success": "#23d160",
  "successInvert": "#fff",
  "text": "#4a4a4a",
  "textInvert": "#fff",
  "textLight": "#7a7a7a",
  "textStrong": "#363636",
  "turquoiseInvert": "#fff",
  "warning": "#ffdd57",
  "warningInvert": "rgba(0, 0, 0, 0.7)",
  "yellowInvert": "rgba(0, 0, 0, 0.7)",
}
`;
